composer/datasets/streaming/download.py

```diff
@@ -116,7 +116,8 @@ def dispatch_download(remote: Optional[str], local: str):
         else:
             raise NotImplementedError
         os.rename(tempfile, local_decompressed)
-        os.remove(local)
+        if local != local_decompressed:
+            os.remove(local)
```

Comment:
Is this still needed?

Reply:
I believe so, otherwise the local file might be deleted as soon as it's downloaded.

Reply:
But I think this is inside the decompression block, so in this block the local file will never be the same as `local_decompressed` right?

Reply:
oh good catch